### ISSUE TITLE
Add a `LogObserver()` function to `Member` interface

### ISSUE THREAD

**Issue by clement2026** [etcd-io/etcd#18519]
### What would you like to be added?

Add a `LogObserver()` function to `Member` interface. 

https://github.com/etcd-io/etcd/blob/4bb9392289623e8bf05916281bd696fbbe42fd4d/tests/framework/interfaces/interface.go#L39-L43

It may look something like this:
```go
 type Member interface { 
 	Client() Client 
 	Start(ctx context.Context) error 
 	Stop() 
        LogObserver()  LogObserver         <<< here
 } 
```

### Why is this needed?

I was writing [a test](https://github.com/etcd-io/etcd/blob/ccbec075e2e180a294935b973d15409a0ea73a8d/tests/integration/raft_log_snapshot_test.go#L60) that waits for events by watching logs. Then I noticed both e2e test and integration test have their own log observer implementations.

https://github.com/etcd-io/etcd/blob/4bb9392289623e8bf05916281bd696fbbe42fd4d/tests/framework/integration/cluster.go#L549-L589

https://github.com/etcd-io/etcd/blob/4bb9392289623e8bf05916281bd696fbbe42fd4d/tests/framework/e2e/etcd_process.go#L45-L69

Would it be a good idea to add a `LogObserver()` function to `Member` interface? This way, I can rewrite the test to work for both e2e and integration tests.

**Comment by serathius** [comment]
I don't think it will be easy as for integration tests you cannot easily collect logs. You need inject a logger that does it for you.
